pylint/lint/caching.py

```diff
@@ -16,7 +16,12 @@
 def _get_pdata_path(
     base_name: Path, recurs: int, pylint_home: Path = Path(PYLINT_HOME)
 ) -> Path:
-    underscored_name = "_".join(str(p) for p in base_name.parts)
```

Comment:
It's going to be hard to backport, this function was in ``pylint.config.__init__.py`` in 2.13 and PYLINT_HOME was a global, there was a ``pathlib.Path`` refactor, there was no test previously... I think we should not backport, especially considering that 2.14.0 will be out in such a short time.

Reply:
I think it's ok to release this with 2.14. Especially since I'm probably one of the few who currently has this problem, as long as the extension is not spreaded more. For 2.13.x would probably anyway https://github.com/microsoft/vscode-pylint/issues/30#issuecomment-1118534288 the solution (even if I had not tested under unix)?

Reply:
For reference the fix in 2.13 is ``base_name = os.path.splitdrive(base_name.replace(os.sep, "_"))[1]``